modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectEntryLocalServiceImpl.java

```diff
@@ -241,6 +241,8 @@ public ObjectEntry addObjectEntry(
 
 		_setExternalReferenceCode(objectEntry, values);
 
+		objectEntry.setExternalReferenceCode(
```

Comment:
The Object Definition has a field named "External Reference Code", which means that every entry (object entry) can have its External Reference Code. There is a private method in this class: _setExternalReferenceCode(). This method searches for this value in the object entry to objectEntry.setExternalReferenceCode(). If there is not, the object entry ID will be the value of the External Reference Code.

Reply:
You can see that this method is being called in line 242.

Reply:
I also believe that the person who opened this bug did not understand this concept, so I think the description is wrong. I will comment there about it.

Reply:
However, this solution would not be right because we cannot set the object definition ERC to its entry.